### DISCUSSION TITLE
Cloud pins backup / share

### DISCUSSION THREAD

**Discussion by seagullua** [organicmaps/organicmaps#627] [Ideas]
It is a handy feature when you need to change your phone a just can lose one. I have recently lost all my Maps.ME pin, which I was collecting during the past few years.

As the map already has import/export of .kmz, the cloud solution is basically file storage. It can also be a good idea to combine with a paid subscription.

Also sharing will be an amazing feature as well. As you can share all pins with people you are traveling together.

**Comment by Jeremboo** [comment]
I saw some open issues related to that requested feature: https://github.com/organicmaps/organicmaps/issues/995 & https://github.com/organicmaps/organicmaps/issues/622

Hope it's something who will comes soon! Already lost many pins when my phone dies and I don't want to lost them again!

On my side, I'll prefer being able to backup them on my own backup, server or platform from Git! But open to any other open source solutions!

**Reply by Unkn0wnCat** [reply]
I think WebDav would be a good call, as it can be used with a lot of self-hosted clouds (e.g. Nextcloud), as well as Storage Spaces (e.g. Hetzner Storagebox), and many NAS-solutions.

WebDav is also easy to implement and would allow Devs to create custom servers, which could enable additional functionality such as sharing.

What do you think?

**Comment by biodranik** [comment]
> As the map already has import/export of .kmz, the cloud solution is basically file storage. It can also be a good idea to combine with a paid subscription.

Simple file storage won't help in some cases:
1. You have two devices, and edited the same bookmarks list on both of them, or edited the same bookmark. What should you see on your both devices after they go online and sync?
2. Imagine you shared your bookmarks with someone else with edit access (I would love to have this feature!). And then we come back to scenario (1) above.

I see that a proper "merge" algorithm can help, but it may be not trivial to implement.

Regarding the storage space, using a self-hosted space is not convenient for most users, so any good ideas are welcome.
